### PR TITLE
Lionelb/refactor api idcc endpoint

### PR DESCRIPTION
**Pull request by lionelB**
refactor idcc searching

### CODE REVIEW

File: packages/code-du-travail-api/src/server/routes/__tests__/idcc.spec.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const request = require("supertest");
+const Koa = require("koa");
+const routes = require("../idcc");
+
+const app = new Koa();
+app.use(routes.routes());
+
+test("return suggest results for demission", async () => {
```
Comment (revolunet) [inline packages/code-du-travail-api/src/server/routes/__tests__/idcc.spec.js]: test description typo ?

**Review by revolunet — APPROVED** [review]

File: packages/code-du-travail-api/src/server/routes/__tests__/idcc.spec.js (hunk 1 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/__tests__/idcc.spec.js]: aha oui  j'update :)

File: packages/code-du-travail-api/src/server/routes/idcc/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const Router = require("koa-router");
+const API_BASE_URL = require("../api").BASE_URL;
+const routeName = "/idcc";
+const router = new Router({ prefix: API_BASE_URL });
+
+const elasticsearchClient = require("../../conf/elasticsearch.js");
+const getIdccBody = require("./idcc.elastic");
+
+const index =
+  process.env.ELASTICSEARCH_DOCUMENT_INDEX || "code_du_travail_numerique";
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/idcc/index.js:10]: Are we duplicating this config ?
We don't have a external `env` config ?

File: packages/code-du-travail-api/src/server/routes/idcc/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Return documents matching the given query.
+ *
+ * @example
+ * http://localhost:1337/api/v1/search?q=incapacité%20travail
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/routes/idcc/index.js]: search ?

**Review by douglasduteil — APPROVED** [review]

File: packages/code-du-travail-api/src/server/routes/idcc/index.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/idcc/index.js:10]: some route/endpoint may use different index name, so i put it on a per routes basis

Comment (douglasduteil) [reply packages/code-du-travail-api/src/server/routes/idcc/index.js:10]: Alright

**Pull request merged**